config/env_config.go

```diff
@@ -363,6 +385,27 @@ func setEndpointDiscoveryTypeFromEnvVal(dst *aws.EndpointDiscoveryEnableState, k
 	return nil
 }
 
+func setUseDualStackEndpointFromEnvVal(dst *aws.DualStackEndpoint, keys []string) error {
+	for _, k := range keys {
+		value := os.Getenv(k)
+		if len(value) == 0 {
+			continue // skip if empty
+		}
+
+		switch {
+		case strings.EqualFold(value, "true"):
+			*dst = aws.DualStackEndpointEnabled
+		case strings.EqualFold(value, "false"):
+			*dst = aws.DualStackEndpointDisabled
+		default:
+			return fmt.Errorf(
+				"invalid value for environment variable, %s=%s, need true, false or auto",
```

Comment:
Doesn't seem like we support `auto`, but error suggests we do

Reply:
I think this was a copy/paste error from another error string. Will fix.